### PR TITLE
docs(eslint-plugin): replaced the Out-of-date TS error codes in eslint-recommended.ts with the supported-codes

### LINKED ISSUE #7582: Docs: Out-of-date TS error codes in eslint-recommended.ts
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

https://github.com/typescript-eslint/typescript-eslint/blob/939e42f05afa9536e03923b8829f060cdf35ada0/packages/eslint-plugin/src/configs/eslint-recommended.ts includes comments referring to corresponding TS error codes for rules. Example:
https://github.com/typescript-eslint/typescript-eslint/blob/939e42f05afa9536e03923b8829f060cdf35ada0/packages/eslint-plugin/src/configs/eslint-recommended.ts#L11C17-L11C17

Some of those error codes are out of date: https://github.com/eslint/eslint/pull/17500#discussion_r1311685415. Someone should go through and update the TS codes to whatever they are now as of the latest stable version of Typescript.

### Affected URL(s)

https://github.com/typescript-eslint/typescript-eslint/blob/939e42f05afa9536e03923b8829f060cdf35ada0/packages/eslint-plugin/src/configs/eslint-recommended.ts

(Technically comments rather than website, but they're used by some as docs too)

### PR DESCRIPTION
**Pull request by girishsontakke**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7582 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview
this PR resolves the issue #7582.

- TS error 2360 was not available in the list of errors at https://github.com/microsoft/TypeScript/blob/v5.2.2/src/compiler/diagnosticMessages.json.
- When trying to negating the left hand operand of [in operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) in this [typescript playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBCBGArGBeGBvAUJgvtglgGYAUAhAEQDWApgJ7kz5hxICUG2OQA) I was getting TS error with code [2322](https://github.com/microsoft/TypeScript/blob/9684ba6b0d73c37546ada901e5d0a5324de7fc1d/src/compiler/diagnosticMessages.json#L1775);

### CODE REVIEW

File: packages/eslint-plugin/src/configs/eslint-recommended.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export = {
         'no-this-before-super': 'off', // ts(2376)
         'no-undef': 'off', // ts(2304)
```
Comment (auvred) [inline packages/eslint-plugin/src/configs/eslint-recommended.ts:24]:
```suggestion
        'no-undef': 'off', // ts(2304) & ts(2552)
```

https://github.com/eslint/eslint/pull/17500#discussion_r1311670674

File: packages/eslint-plugin/src/configs/eslint-recommended.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export = {
         'no-this-before-super': 'off', // ts(2376)
```
Comment (auvred) [inline packages/eslint-plugin/src/configs/eslint-recommended.ts:23]:
```suggestion
        'no-this-before-super': 'off', // ts(2376) & ts(17009)
```
https://github.com/eslint/eslint/pull/17500#discussion_r1311669537
[playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAECCbQN4ChkF9WkjAQgI2gFMAPAF0IDsATGeJZaR6AMwHtWAuaCUgJwEsKAcwZNgrCj14BXYKVa8AFAEp6TddFIALfhAB0bVtAC80AORgzojd2kAHQkuXXoGDMixRoAYWBEylDRwCCjq4pJ8svJKvIRgVBIgAJ4s7FxSgkKqoTaaOvqGLuoQ9o4qLm5AA)

**Review by auvred — CHANGES_REQUESTED** [review]
Please update the other codes according to the existing comments on this PR https://github.com/eslint/eslint/pull/17500

---

- https://github.com/typescript-eslint/typescript-eslint/blob/4899b95bf31d42358d95d078e40424ea95934d16/packages/eslint-plugin/src/configs/eslint-recommended.ts#L17

  https://github.com/eslint/eslint/pull/17500#discussion_r1311644449

- https://github.com/typescript-eslint/typescript-eslint/blob/4899b95bf31d42358d95d078e40424ea95934d16/packages/eslint-plugin/src/configs/eslint-recommended.ts#L18

  https://github.com/eslint/eslint/pull/17500#discussion_r1311657506

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
What @auvred said 🙂

**Comment by girishsontakke** [comment]
Thanks @auvred for the review and the help 🙏. I will add the changes.

**Review by auvred — APPROVED** [review]
Nice! I guess that's it

**Pull request merged**